src/parse.c

```diff
@@ -10467,10 +10469,145 @@ static tree_t p_psl_directive(void)
    return t;
 }
 
-static tree_t p_psl_clock_declaration(void)
+static type_t p_psl_subtype_indication(void)
+{
+   BEGIN("PSL subtype indication");
+
+   // Handle ambiguity in "p_subtype_indication". Two consecutive IDs
+   // may indicate resolution function, or only "type_mark" followed by
+   // actual parameter name.
+   token_t third = peek_nth(3);
+   if (peek() == tID && peek_nth(1) == tID &&
+       (third == tSEMI || third == tCOMMA || third == tRPAREN))
+      return p_type_mark();
+   else
+      return p_subtype_indication();
+}
+
+static type_t p_psl_param_spec(psl_node_t node, psl_type_t *psl_type, class_t *class)
+{
+   //    const
+   //    | [const | mutable] Value_Parameter
+   //    | sequence
+   //    | property
+   //
+   //    Value_Parameter ::=
+   //       HDL_Type
+   //     | PSL_Type_Class
+   //
+   //    HDL_Type ::=
+   //       hdltype HDL_VARIABLE_TYPE
+   //
+   //    PSL_Type_Class ::=
+   //       boolean | bit | bitvector | numeric | string
+
+   BEGIN("PSL Parameter specification");
+
+   *class = C_MUTABLE;
```

Comment:
Is mutable the same or similar to the `variable` class in VHDL?

Reply:
TBH, I am not sure.

Can a parameter to PSL property or sequence be VHDL `signal` and also `shared variable`?
If both are allowed, I would keep the `C_MUTABLE` since it can represent both.
If only `signal` is allowed, then I think using `signal` class would be intuitive.

Or, does `variable` class represent anything that is "changable", and not kind of VHDL parameter ?

Reply:
I don't think that other object classes beside `signal` make sense as parameters. But I'm not sure if it is defined exactly.

Reply:
I would expect that if it is not specified, passing shared variable could be supported, but I agree that it is rather exotic case (possibly with many corner-cases), so I changed it by default to `C_SIGNAL`.